addons/mail/static/src/core/common/channel_command_model.js

```diff
@@ -0,0 +1,42 @@
+import { Record } from "@mail/core/common/record";
+
+export class ChannelCommand extends Record {
+    static id = "name";
+    /** @type {Object.<number, import("models").ChannelCommand>} */
+    static records = {};
+    /** @returns {import("models").ChannelCommand} */
+    static get(data) {
+        return super.get(data);
+    }
+    /** @returns {import("models").ChannelCommand|import("models").ChannelCommand[]} */
+    static insert(data) {
+        return super.insert(...arguments);
+    }
+
+    /** @type {number} */
+    endPosition;
+    /** @type {string} */
+    methodName;
+    /** @type {string} */
+    name;
+    /** @type {Object} */
+    subCommandData = {};
+    /** @type {string[]} */
+    subCommandFields;
```

Comment:
It does not look clean to me to rely on such an arbitrary difference to determine if the command has sub-commands. A dedicated field does not cost much but greatly simplifies understanding the sub-command mechanism, especially since you already defined a getter. Also, from what I understand, the reason is that we *might* need it later. I don't think planning for potential future needs leads to good decisions. Also, it's not consistent with `subCommandData`.